reana_commons/utils.py

```diff
@@ -218,9 +264,28 @@ def get_disk_usage(directory, summarize=False, to_human_readable_units=None):
     if "Darwin" not in platform.system():
         # Default block size in GNU is KB
         command.append("-b")
-    command.append(absolute_path)
-    disk_usage_info = subprocess.check_output(command).decode().split()
-    # create pairs of (size, filename)
+
+    name_filter = None
+    size_filter = None
+    if search:
+        search = json.loads(search)
+        name_filter = search.get("name")
+        size_filter = search.get("size")
+
+    if name_filter:
+        path_list = []
+        for _path in name_filter:
+            path_list += _get_files_recursive_wildcard(
+                absolute_path.decode("utf-8"), _path
+            )
+        if path_list:
+            for path in path_list:
+                command.append(path)
+        disk_usage_info = subprocess.check_output(command).decode().split()
```

Comment:
Minor: perhaps this could be moved to an internal function as well? Separation could be useful for someone modifying the filtering code in the future